test/Microsoft.AspNet.Mvc.Razor.Test/RazorPageTest.cs

```diff
@@ -687,96 +687,232 @@ public void IsSectionDefined_ThrowsIfPreviousSectionWritersIsNotRegistered()
+            }
 
             // Act
             var page = CreatePage(p =>
             {
-                p.Write((ITextWriterCopyable)contentToBeCopied);
+                p.HtmlEncoder = new HtmlEncoder();
+                p.WriteTagHelperAsync(tagHelperExecutionContext).Wait();
             }, context);
             await page.ExecuteAsync();
 
             // Assert
-            Assert.Equal(expectedContent, writer.ToString());
+            Assert.Equal(expected, writer.ToString());
         }
 
         [Fact]
-        public async Task WriteTo_ITextWriterCopyable_WritesContent_ToSpecifiedWriter()
+        public async Task WriteTagHelperToAsync_WritesToSpecifiedWriter()
         {
             // Arrange
-            var writer = new StringWriter();
-            var expectedContent = "Hello World!";
-            var contentToBeCopied = new DefaultTagHelperContent().SetContent("Hello ").Append("World!");
+            var writer = new StringCollectionTextWriter(Encoding.UTF8);
+            var context = CreateViewContext(new StringWriter());
+            var tagHelperExecutionContext = new TagHelperExecutionContext(tagName: "p",
+                   selfClosing: false,
+                   items: new Dictionary<object, object>(),
+                   uniqueId: string.Empty,
+                   executeChildContentAsync: () => { return Task.FromResult(result: true); },
+                   startTagHelperWritingScope: () => { },
+                   endTagHelperWritingScope: () => new DefaultTagHelperContent());
+            tagHelperExecutionContext.Output =
+                new TagHelperOutput("p", new Dictionary<string, string>());
 
             // Act
             var page = CreatePage(p =>
             {
-                p.WriteTo(writer, (ITextWriterCopyable)contentToBeCopied);
-            });
+                p.WriteTagHelperToAsync(writer, tagHelperExecutionContext).Wait();
+            }, context);
             await page.ExecuteAsync();
 
             // Assert
-            Assert.Equal(expectedContent, writer.ToString());
+            Assert.Equal("Hello World!", writer.ToString());
         }
 
-        [Fact]
-        public async Task Write_TagHelperContent_WritesContent()
+        [Theory]
+        [MemberData(nameof(WriteTagHelper_InputData))]
+        public async Task WriteTagHelperToAsync_WritesFormattedTagHelper(TagHelperOutput output, string expected)
         {
             // Arrange
-            // This writer uses BufferEntryCollection underneath and so can copy the buffer.
             var writer = new StringCollectionTextWriter(Encoding.UTF8);
-            var context = CreateViewContext(writer);
-            var expectedContent = "Hello World!";
-            var contentToBeCopied = new DefaultTagHelperContent().SetContent("Hello ").Append("World!");
+            var context = CreateViewContext(new StringWriter());
+            var tagHelperExecutionContext = new TagHelperExecutionContext(tagName: output.TagName,
```

Comment:
nit: had to look around to find the first argument.  it should also be on its own line.